File: erezervisi_api/eRezervisi/eRezervisi.Common.Dtos/Auth/JwtTokenResponse.cs
namespace eRezervisi.Common.Dtos.Auth
{
    public class JwtTokenResponse
    {
        public bool LoggedIn { get; set; }
        public string? Token { get; set; }
        public DateTime? ExpiresAtUtc { get; set; }
        public string? RefreshToken { get; set; }
        public DateTime? RefreshTokenExpiresAtUtc { get; }
        public long RoleId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Initials => $"{FirstName.First()}{LastName.First()}";
        public string Email { get; set; }
        public string Username { get; set; }
        public long UserId { get; set; }
        public string? Image { get; set; }
        public bool ReceiveNotifications { get; set; }

        public JwtTokenResponse(bool loggedIn, string? token, DateTime? expiresAtUtc, string? refreshToken,
            DateTime? refreshTokenExpiresAtUtc, long roleId, string firstName, string lastName,
            string email, string username, long userId, string? image, bool receiveNotifications)
        {
            LoggedIn = loggedIn;
            Token = token;
            ExpiresAtUtc = expiresAtUtc;
            RefreshToken = refreshToken;
            RefreshTokenExpiresAtUtc = refreshTokenExpiresAtUtc;
            RoleId = roleId;
            FirstName = firstName;
            LastName = lastName;
            Email = email;
            Username = username;
            UserId = userId;
            Image = image;
            ReceiveNotifications = receiveNotifications;
        }
    }
}

File: erezervisi_desktop/erezervisi_desktop/lib/models/responses/auth/jwt_token_response.dart
class JwtTokenResponse {
  final bool loggedIn;
  final String token;
  final DateTime expiresAtUtc;
  final String refreshToken;
  final DateTime refreshTokenExpiresAtUtc;
  final num roleId;
  final String firstName;
  final String lastName;
  final String initials;
  final String email;
  final String username;
  String? image;
  final num userId;
  final bool receiveNotifications;

  JwtTokenResponse(
      {required this.loggedIn,
      required this.token,
      required this.expiresAtUtc,
      required this.refreshToken,
      required this.refreshTokenExpiresAtUtc,
      required this.roleId,
      required this.firstName,
      required this.lastName,
      required this.initials,
      required this.email,
      required this.username,
      this.image,
      required this.userId,
      required this.receiveNotifications});

  factory JwtTokenResponse.fromJson(Map<String, dynamic> json) {
    return JwtTokenResponse(
        loggedIn: json['loggedIn'] ?? false,
        token: json['token'] ?? '',
        expiresAtUtc: DateTime.parse(json['expiresAtUtc']),
        refreshToken: json['refreshToken'] ?? '',
        refreshTokenExpiresAtUtc:
            DateTime.parse(json['refreshTokenExpiresAtUtc']),
        roleId: json['roleId'] ?? 0,
        firstName: json['firstName'] ?? '',
        lastName: json['lastname'] ?? '',
        initials: json['initials'] ?? '',
        email: json['email'] ?? '',
        username: json['username'] ?? '',
        image: json['image'],
        receiveNotifications: json['receiveNotifications'],
        userId: json['userId'] ?? 0);
  }

  Map<String, dynamic> toJson() {
    return {
      'loggedIn': loggedIn,
      'token': token,
      'expiresAtUtc': expiresAtUtc.toIso8601String(),
      'refreshToken': refreshToken,
      'refreshTokenExpiresAtUtc': refreshTokenExpiresAtUtc.toIso8601String(),
      'roleId': roleId,
      'firstName': firstName,
      'lastName': lastName,
      'initials': initials,
      'email': email,
      'image': image,
      'username': username,
      'receiveNotifications': receiveNotifications,
      'userId': userId
    };
  }
}
